server/events/events.go

```diff
@@ -28,6 +29,19 @@ type Clientlike interface {
 	Info() Client
 }
 
+// OnProcessMessage function is called right after a publish message is received.
```

Comment:
> OnProcessMessage function is called right after a publish message is received. It is called right after ACL checking and before any other data gets evaluated for publishing. So this event e.g. allows changing the Retain flag.

Overall a great comment 👍🏻 I feel like this opener could be a little clearer - maybe something like:

> OnProcessMessage is called when a publish message is received, allowing modification of the packet data after ACL checking has occurred but before any data is evaluated for processing - e.g. for changing the Retain flag.